staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -521,53 +540,62 @@ func (rdm *resourceDiscoveryManager) serveHTTP(resp http.ResponseWriter, req *ht
 	response := cache.cachedResponse
 	etag := cache.cachedResponseETag
 
-	mediaType, _, err := negotiation.NegotiateOutputMediaType(req, rdm.serializer, DiscoveryEndpointRestrictions)
+	writeDiscoveryResponse(&response, etag, rdm.serializer, resp, req)
+}
+
+func writeDiscoveryResponse(
```

Comment:
Is introducing this function needed in this PR?  I looks to me like it would fine to leave this code in `serveHTTP`, and would make it sightly easier to review this PR (any non-gate changes need to be reviewed VERY carefully, and this adds noise)

Reply:
Ah, I missed the new call site. nvm.